Makefile

```diff
@@ -112,6 +97,10 @@ black:
 isort:
 	$(PYTHON) -m isort --check src tests docs/examples
 
+.PHONY: ruff
+ruff:
+	$(PYTHON) -m ruff --preview src tests docs/examples
```

Comment:
**praise**: very nice. There was at least a couple of times I wanted to run `make ruff` and couldn't :)